bats/tests/containers/published-ports.bats

```diff
@@ -0,0 +1,35 @@
+load '../helpers/load'
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'start container engine' {
+    start_container_engine
+    wait_for_container_engine
+}
+
+run_container_with_published_port() {
+    ctrctl pull "$IMAGE_NGINX"
+    ctrctl run -d -p "$@" --restart=no "$IMAGE_NGINX"
+}
+
+verify_container_published_port() {
+    run try --max 9 --delay 10 curl --insecure --silent --show-error "$@"
+    assert_success
+    assert_output --partial 'Welcome to nginx!'
+}
+
+@test 'container published port binding on localhost' {
+    run_container_with_published_port "127.0.0.1:8080:80"
+    verify_container_published_port "http://127.0.0.1:8080"
+    skip_unless_host_ip
+    run curl --head "http://${HOST_IP}:8080"
+    assert_output --partial "curl: (7) Failed to connect"
```

Comment:
I think this part needs to be in a separate `@test` block. We want to see a pass result for the code above, and not have it hidden because we don't have a host ip.